tests/ui/drop/lint-if-let-rescope.stderr

```diff
@@ -0,0 +1,73 @@
+error: `if let` assigns a shorter lifetime since Edition 2024
+  --> $DIR/lint-if-let-rescope.rs:25:8
+   |
+LL |     if let Some(_value) = droppy().get() {
+   |        ^^^^^^^^^^^^^^^^^^^--------^^^^^^
+   |                           |
+   |                           this value has a significant drop implementation which may observe a major change in drop order and requires your discretion
+   |
+   = warning: this changes meaning in Rust 2024
+   = note: for more information, see issue #124085 <https://github.com/rust-lang/rust/issues/124085>
+help: the value is now dropped here in Edition 2024
+  --> $DIR/lint-if-let-rescope.rs:29:6
+   |
+LL |     } else {
+   |      ^
```

Comment:
Nit: the span looks a lil strange here, it's like off-by-one?

Reply:
I am subtracting `BytePos(1)` from the span `hi` now. I haven't found any better way, but we can safely say that we exclusively use ASCII `}`.

Reply:
The off-by-one is now fixed with `SourceMap::end_point`.